utils/src/treev2/quickPickWizard/experiences/compatibility/compatibilityPickSubscriptionExperience.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as types from '../../../../../index';
+import * as vscode from 'vscode';
+import { ISubscriptionContext } from '@microsoft/vscode-azext-dev';
+import { subscriptionExperience } from '../subscriptionExperience';
+import { createSubscriptionContext } from '../../../../utils/credentialUtils';
+import { ResourceGroupsItem } from '../../quickPickAzureResource/tempTypes';
+import { isAzExtTreeItem } from '../../../../tree/isAzExtParentTreeItem';
+
+/**
+ * Returns `ISubscriptionContext` instead of `ApplicationSubscription` for compatibility.
+ */
+export async function compatibilitySubscriptionExperience(context: types.IActionContext, tdp: vscode.TreeDataProvider<ResourceGroupsItem>): Promise<ISubscriptionContext> {
```

Comment:
Nit: I feel like we need a name that's a little more verb-y than `compatibilityXxxExperience()`, maybe something plain and boring like `pickSubscriptionWithCompatibility()`?

Reply:
I organized it into a namespace so it's `PickTreeItemWithCompatibility.subscription`